cmake/extensions.cmake

```diff
@@ -2261,3 +2263,368 @@ function(target_byproducts)
                      COMMENT "Logical command for additional byproducts on target: ${TB_TARGET}"
   )
 endfunction()
+
+########################################################
+# 4. Zephyr devicetree function
+########################################################
+# 4.1. dt_*
+#
+# The following methods are for retrieving devicetree information in CMake.
+#
+# Note: In CMake we refer to the nodes using the node's path, therefore there
+# is no dt_path(...) function for obtaining a node identifier.
+
+# Usage:
+#   dt_nodelabel(<var> LABEL <label>)
+#
+# function for retrieving the node path for the node having nodelabel
+# <label>.
+#
+# the node path will be returned in the <var> parameter.
```

Comment:
what is returned if <label> doesn't exist?

Reply:
the the return variable will be undefined, and you can test it like:
```
dt_nodelabel(nodepath LABEL mylabel)
if(DEFINED nodepath)
  # I got a node path, do something
endif()
```

Reply:
> the the return variable will be undefined, and you can test it like:

Same here, I think the request is for documented behavior.

Reply:
Function description improved